src/modules/scraper/adapter/messaging/TrackFillEventConsumer.ts

```diff
@@ -0,0 +1,167 @@
+import { OnQueueFailed, Process, Processor } from "@nestjs/bull";
+import { Logger } from "@nestjs/common";
+import { Job } from "bull";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { utils } from "ethers";
+import BigNumber from "bignumber.js";
+import { DateTime } from "luxon";
+
+import { TrackFillEventQueueMessage, ScraperQueue } from ".";
+import { Deposit } from "../../model/deposit.entity";
+import { TrackService } from "../amplitude/track-service";
+import {
+  deriveRelayerFeeComponents,
+  fixedPointAdjustment,
+  makeAmountValuesFormatter,
+  makeWeiPctValuesFormatter,
+} from "../amplitude/utils";
+import { EthProvidersService } from "../../../web3/services/EthProvidersService";
+import { chainIdToInfo } from "../../../../utils";
+import { MarketPriceService } from "../../../market-price/services/service";
+
+@Processor(ScraperQueue.TrackFillEvent)
+export class TrackFillEventConsumer {
+  private logger = new Logger(TrackFillEventConsumer.name);
+
+  constructor(
+    private providers: EthProvidersService,
+    private marketPriceService: MarketPriceService,
+    private trackService: TrackService,
+    @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
+  ) {}
+
+  @Process()
+  private async process(job: Job<TrackFillEventQueueMessage>) {
+    if (!this.trackService.isEnabled()) {
+      this.logger.verbose("Amplitude tracking is disabled");
+      return;
+    }
+
+    const { depositId, fillTxHash, destinationToken } = job.data;
+    const deposit = await this.depositRepository.findOne({ where: { id: depositId }, relations: ["token", "price"] });
+
+    if (!deposit) {
+      this.logger.verbose("Deposit not found in db");
+      return;
+    }
+
+    if (!deposit.token || !deposit.price || !deposit.depositDate) {
+      throw new Error("Can not track fill event without token, price or deposit date");
+    }
+
+    const fillTx = deposit.fillTxs.find((tx) => tx.hash === fillTxHash);
+
+    if (!fillTx) {
+      throw new Error("Fill tx does not exist on deposit");
+    }
+
+    if (!fillTx.date) {
+      throw new Error("Fill tx does not have a date");
+    }
+
+    const destinationChainInfo = chainIdToInfo[deposit.destinationChainId];
+    const sourceChainInfo = chainIdToInfo[deposit.sourceChainId];
```

Comment:
I think we need to handle the case when sourceChainInfo and destinationChainInfo are `undefined`

Reply:
Yea good point. I will add a change that defaults to `"-"` if chain info is unknown.

Reply:
https://github.com/across-protocol/scraper-api/pull/186/commits/a310e99351d2e2e628674b4706d311c914965312